### PR TITLE
FIXED: Refactoring Request Headers

### PR DESCRIPTION
**Pull request by AnthonyJCampbell**
# Description
Normalizing all Axios requests to use the correct request handlers and have homogenous headers. This also included the refactoring of in-component Axios requests and adding some extra functionality to the setHeaders utility function to incorporate token-retrieval.

## Checklist
[x] I have performed a self-review of my own code
[x] I have commented my code, particularly in hard-to-understand areas
[x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: src/components/pages/UpdateFarmer/UpdateFarmer.js [hunk 1]
```diff
@@ -184,35 +179,28 @@ const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
...
+        // Added "_id" because formData doesn't have/need an _id property
+        farmer: { ...formData, _id: farmerData._id }
       });
+    })
+    .catch(err => {
+      err.response.data.errors.forEach(element => {
+        toast.error(element.message);
+      });
+      return;
+    });
```
Comment (Pav0l) [inline src/components/pages/UpdateFarmer/UpdateFarmer.js]: This catch block needs to stay as it is coming from `master`.

File: src/components/pages/UpdateFarmer/UpdateFarmer.js [hunk 2]
```diff
@@ -22,7 +20,7 @@ const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
 
     // eslint-disable-next-line no-unused-vars
     for (const inputSection in formInputData) {
-      const inputSectionData = formInputData[inputSection];
+      const inputSectionData = formInputData[inputSection];
```
Comment (Pav0l) [inline src/components/pages/UpdateFarmer/UpdateFarmer.js]: 👾 👩‍🚀  space 👨‍🚀 🛰

**Comment by Pav0l** [comment]
[Refactor Axios Handlers](https://trello.com/c/Jc4etI7w/127-refactor-axios-handlers)

File: src/components/pages/UpdateFarmer/UpdateFarmer.js [hunk 3]
```diff
@@ -135,6 +133,7 @@ const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
             );
           }
         } catch (error) {
+          setStateLoading(false);
```
Comment (quansenB) [inline src/components/pages/UpdateFarmer/UpdateFarmer.js]: good catch

File: src/components/pages/AddStaff/AddStaff.js [hunk 4]
```diff
@@ -3,7 +3,6 @@ import { useState } from 'react';
 
 import {
   registrationHandler,
```
[the remark this answers is not in the document]
Comment (Pav0l) [reply src/components/pages/AddStaff/AddStaff.js]: No, but good catch 👍

File: src/components/pages/UpdateFarmer/UpdateFarmer.js [hunk 5]
```diff
@@ -135,6 +133,7 @@ const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
...
           toast.error('Failed to upload image. Please check your connection.');
 
           // Display error message in the console for context
```
[the remark this answers is not in the document]
Comment (Pav0l) [reply src/components/pages/UpdateFarmer/UpdateFarmer.js]: This is not in scope of this PR. We can look into in in another PR that will handle error handling

**Pull request merged**